sdk/python/kfp/compiler/compiler.py

```diff
@@ -140,9 +145,9 @@ def _op_to_template(self, op):
       template['nodeSelector'] = op.node_selector
 
     if op.env_variables:
-      template['container']['env'] = list(map(self._convert_k8s_obj_to_dic, op.env_variables))
+      template['container']['env'] = list(map(K8sHelper.convert_k8s_obj_to_json, op.env_variables))
```

Comment:
The output in not really JSON. It's a python dict.
P.S. The function (including the name comes from the K8s client library (`ApiClient` class))